### PR TITLE
CMake: new wxUSE_PEDANTIC_FLAGS option has been added

### PR DESCRIPTION
**Pull request by Deamhan**
It can be useful for possible errors detection.

### CODE REVIEW

**Comment by Deamhan** [comment]
Some modifications have been added for clang case.

File: build/cmake/options.cmake [hunk 1]
```diff
@@ -44,6 +44,8 @@ if(WIN32)
     set(wxBUILD_VENDOR "custom" CACHE STRING "Short string identifying your company (used in DLL name)")
 endif()
 
+wx_option(wxUSE_PEDANTIC_FLAGS, "use additional warning/error compiler flags" OFF)
```
Comment (MaartenBent) [inline build/cmake/options.cmake]:
```suggestion
wx_option(wxUSE_PEDANTIC_FLAGS "use additional warning/error compiler flags" OFF)
```

**Review by MaartenBent — COMMENTED** [review]
Is this really useful? I already get thousands of warnings when compiling wxzlib and wxjpeg with clang (including `-Wdisabled-macro-expansion`, `-Wreserved-id-macro`, `-Wpadded`).

Instead of `Wall`, `Wextra` end `Weverything`, it might be better to specifically enable checks for warnings we care about. And only add the warnings for wxWidgets libraries, not external ones (similar as `target_compile_options(${target_name} PRIVATE /W4)` in `functions.cmake`).

And it finishes with:
`warning: unknown warning option '-Wstack-usage=16384'; did you mean '-Wtype-safety'? [-Wunknown-warning-option]`
`warning: unknown warning option '-Werror=stringop-overflow'; did you mean '-Werror=shift-overflow'? [-Wunknown-warning-option]`

**Comment by Deamhan** [comment]
This thing can be adjusted, but if it is not useful. I didn't check all options on clang, I use it in GCC case. It was just offering and if it is not very useful for others we should do something more useful, for example add some options from configure to cmake.

**Pull request closed without merging**